solutions/day_3/solution.py

```diff
@@ -0,0 +1,42 @@
+from typing import List
+from functools import reduce
+
+TEXT_FILE_LOCATION = "input.txt"
+LOWER_CASE_ORD_DIFFERENCE = 96
+UPPER_CASE_ORD_DIFFERENCE = 38
+
+
+def find_intersection(acc: List, val: List[str]) -> List[int]:
+    if val != "":
+        midpoint = len(val) // 2
+        char_to_add = set(val[:midpoint]).intersection(set(val[midpoint:])).pop()
+        if char_to_add.islower():
+            acc.append(ord(char_to_add) - LOWER_CASE_ORD_DIFFERENCE)
+        else:
+            acc.append(ord(char_to_add) - UPPER_CASE_ORD_DIFFERENCE)
+    else:
+        acc += []
+    return acc
+
+
+def create_groups(acc: List[List], val: List[str]) -> List[int]:
+    char_to_add = set(val[0]).intersection(set(val[1]).intersection(set(val[2]))).pop()
+    if char_to_add.islower():
+        acc.append(ord(char_to_add) - LOWER_CASE_ORD_DIFFERENCE)
+    else:
+        acc.append(ord(char_to_add) - UPPER_CASE_ORD_DIFFERENCE)
+    return acc
+
+
+def main():
+    with open(TEXT_FILE_LOCATION) as file:
+        lines = list(map(lambda s: s.rstrip("\n"), file.readlines()))
+    elf_item_list = reduce(find_intersection, lines, [])
+    final = list(zip(*[iter(lines)] * 3))
```

Comment:
Really goddamn clever! I really like it. I'd add a comment explaining how it works, because it is not the most obvious, but otherwise, great stuff!